File: 01-week-1--starter-algorithms/00-day-1--reverse-a-string/javascript/reverse_string.js
function reverseString(str) {
  let backwardsWord  = ""
  for (let i = str.length - 1; i > -1; --i){
    backwardsWord = backwardsWord + str[i]
  }
  return backwardsWord
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("hi"));

  console.log("");

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));

  console.log("");

  console.log("Expecting: 'tacocat ekil'");
  console.log("=>", reverseString("like tacocat"));
}

module.exports = reverseString;

// Please add your pseudocode to this file
// And a written explanation of your solution

//Pseudocode: 
// need to iterate through index and replace - index[0] becomes index[str-1] and so on

//Explanation: 
//iterate through string, put each character in turn into result string

File: 01-week-1--starter-algorithms/00-day-1--reverse-a-string/ruby/reverse_string.rb
def reverse_string(str)
  backwards_string = ""
  str.chars.each do |letter|
    backwards_string = letter + backwards_string
  end
  backwards_string
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts "=>", reverse_string('hi')

  puts

  puts "Expecting: 'ybabtac'"
  puts "=>", reverse_string('catbaby')

  # Don't forget to add your own!
  puts

  puts "Expecting: 'tacocat ekil"
  puts "=>", reverse_string('like tacocat')
end

# Please add your pseudocode to this file
# And a written explanation of your solution

# Pseudocode:
# iterate over each letter, add to a string, then add up

# Explanation: 
# Take each letter, store it, then as go through string will be stored in reverse